doc/BSA Node.md

```diff
@@ -1,33 +1,72 @@
 
-The current BSA module returns the BSA results all together when it receives the last photon from either  neighbor.
-Once the result is received by the neighbor, if any, the neighbor terminates the emission, and resets all qubits that failed entangling. If the BSA receives the last qubit from right, and generates/sends the classical BSA results packet,  but if another last qubit from left arrives because it had a larger buffer size, the BSA needs to know that he already sent back the results for this trial - it needs to correctly ignore the last qubit from left. In the simulation, I have avoided this problem by tracking the first photon (pulse indicating the start of photon burst). The boolean "this_trial_done" becomes true if the last qubit from either neighbor has been detected. While "this_trial_done == true", any BSA result will be dumped. "this_trial_done" will be set false when the BSA detects the first photon from either qubit.
-What's important is that the BSA nodes need to recognize the beginning and the end of each trial, both of them!!!
+![Beam Splitter (4).png](./img/Beam_Splitter_(4).png)
+
+Here, we assume those entangled states are <img src="https://render.githubusercontent.com/render/math?math=|\psi^{-}\rangle = \frac{|HV\rangle - |VH\rangle}{\sqrt{2}}"> known as one of the Bell states, where <img src="https://render.githubusercontent.com/render/math?math=|H\rangle"> is horizontal polarization state and <img src="https://render.githubusercontent.com/render/math?math=|V\rangle"> is vertical polarization state. 
+
+Beam splitter (or semi-reflective mirror) is one optical component that splits incident light into two, transmitted and reflected light. 
+
+![BSA (3).png](./img/BSA_D.png)
+
+When there are joint detection in (D1, D3) or (D2, D4), the remaining qubit's state<img src="https://render.githubusercontent.com/render/math?math=|\psi^{-}\rangle">. If there are joint detection in (D1, D2) or (D3, D4), the state is <img src="https://render.githubusercontent.com/render/math?math=|\psi^{+}\rangle">.  In the case where both photons are detected at one detector, the state is <img src="https://render.githubusercontent.com/render/math?math=|\phi^{+}\rangle"> or <img src="https://render.githubusercontent.com/render/math?math=|\phi^{-}\rangle">.  It is impossible to distinguish either <img src="https://render.githubusercontent.com/render/math?math=|\phi^{+}\rangle"> or <img src="https://render.githubusercontent.com/render/math?math=|\phi^{-}\rangle">. From this fact,  the maximum success probability of linear optical Bell state measurement is known as theoretically limited to 50 %[4].
+
+# What is the interface to the outside world?
+
+BSA provides usually two interfaces through fiber or laser to take in two photons. 
+
+# Who interacts with this interface?
+
+There are two cases.
+
+1. BSA is inside one of the nodes. (MM link)
```

Comment:
Here, we are possibly confusing the issue of the BSA *Node* (which is right there in the name of the .md file) and the BSA (or HOM) functionality.